src/baseclasses/chemical_energy/nome_cp_analysis.py

```diff
@@ -0,0 +1,145 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+
+from nomad.metainfo import Quantity, Reference, Section, SubSection, Datetime
+from nomad.datamodel.metainfo.basesections import Analysis, SectionReference, AnalysisResult, CompositeSystemReference
+from nomad.datamodel.results import Results, Material
+
+from baseclasses.chemical_energy import Chronopotentiometry
+from ..helper.utilities import get_reference
+
+
+class CPOERReference(SectionReference):
+    reference = Quantity(
+        type=Reference(Chronopotentiometry.m_def),
+        a_eln=dict(component='ReferenceEditQuantity', label='OER Chronopotentiometry Measurement'))
+
+
+class CPOERAnalysisResult(AnalysisResult):
+    voltage_avg_first5 = Quantity(
+        type=np.dtype(np.float64),
+        unit=('V'))
+    voltage_avg_last5 = Quantity(
+        type=np.dtype(np.float64),
+        unit=('V'))
+    voltage_difference = Quantity(
+        type=np.dtype(np.float64),
+        unit=('V'))
+    j = Quantity(
+        type=np.dtype(np.float64),
+        description='current density',
+        unit=('A/cm^2'))
+    current_density_string = Quantity(
+        type=str,
+        description='string representation of j (needed for terms in explore view)',)
+    experiment_duration = Quantity(
+        type=np.dtype(np.float64),
+        unit=('hr'))
+    reaction_type = Quantity(
+        type=str,
+        description='At the moment only OER CP is supported. In the future maybe also NRR CP.')
+    samples = SubSection(
+        section_def=CompositeSystemReference,
+        repeats=True,)
+
+    def normalize(self, archive, logger):
+        self.current_density_string = format(self.j, '~')
+        super(CPOERAnalysisResult, self).normalize(archive, logger)
+
+
+class CPAnalysis(Analysis):
+    m_def = Section(label_quantity='name')
+
+    inputs = Analysis.inputs.m_copy()
+    inputs.section_def = CPOERReference
+
+    outputs = Analysis.outputs.m_copy()
+    outputs.section_def = CPOERAnalysisResult
+
+    def normalize(self, archive, logger):
+        refs = find_all_cp_in_upload(archive, archive.metadata.upload_id)
+        self.inputs = [CPOERReference(reference=ref) for ref in refs]
+
+        if self.inputs is not None and len(self.inputs) >= 2:
+            first_oer_run = self.inputs[0].reference
+            last_oer_run = self.inputs[-1].reference
+
+            for cp_entry in self.inputs:
```

Comment:
here you sort, you can already sort when you get the references.

Reply:
:white_check_mark: